From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Softik Lord <dimap9986@gmail.com>
Date: Tue, 12 Jul 2022 00:13:11 +0500
Subject: [PATCH] Swaps-the-predicate-order-of-collision


diff --git a/src/main/java/net/minecraft/server/Entity.java b/src/main/java/net/minecraft/server/Entity.java
index ae3065fb234f9011b600ca00ec464856d59028f4..ac1df230416a6cb94241f2a60e5e19930319ba48 100644
--- a/src/main/java/net/minecraft/server/Entity.java
+++ b/src/main/java/net/minecraft/server/Entity.java
@@ -1408,8 +1408,9 @@ public abstract class Entity implements ICommandListener, KeyedObject { // Paper
     public void d(EntityHuman entityhuman) {}
 
     public void collide(Entity entity) {
+        if (entity.noclip || this.noclip) return; // NeonPaper - Test this earlier
         if (!this.x(entity)) {
-            if (!entity.noclip && !this.noclip) {
+            if (true || !entity.noclip && !this.noclip) { // NeonPaper - Moved up
                 double d0 = entity.locX - this.locX;
                 double d1 = entity.locZ - this.locZ;
                 double d2 = MathHelper.a(d0, d1);
